include/singa/core/memory.h

```diff
@@ -50,6 +51,57 @@ class DeviceMemPool {
 //  size_t init_size_ = 0, max_size_ = 0;
 };
 
+class CppMemPool {
+	public:
+		// initial pool size (MB), and the size of each memory uint in the memory pool (KB)
+		CppMemPool(size_t init_size_mb = 256, size_t uint_size_kb = 1);
```

Comment:
optimizations could be added later, e.g.
OptimizeMemPool():  re-compute the total size blocks in use, and move them into contiguous memory space.